### PR TITLE
perf(migrate): migrate Aad config with skipProvision

### PR DESCRIPTION
**Pull request by wenytang-ms**

### CODE REVIEW

**Review by KennethBWSong — DISMISSED** [review]
LGTM

File: packages/fx-core/tests/core/hooks.test.ts [hunk 1]
```diff
@@ -632,6 +632,7 @@ describe("Middleware - others", () => {
       await fs.ensureDir(projectPath);
       mockedEnvRestore = mockedEnv({
         TEAMSFX_INSIDER_PREVIEW: "true",
+        AAD_APP_CLIENT_SECRET: "test",
```
Comment (blackchoey) [inline packages/fx-core/tests/core/hooks.test.ts:635]: do we need to set this environment variable?

File: packages/fx-core/src/core/middleware/projectMigrator.ts [hunk 2]
```diff
@@ -480,6 +486,27 @@ async function migrateMultiEnv(projectPath: string): Promise<void> {
       configDev["skipAddingSqlUser"] =
         envDefault[ResourcePlugins.AzureSQL][EnvConfigName.SqlSkipAddingUser];
     }
+    if (envDefault[ResourcePlugins.Aad]?.[EnvConfigName.AadSkipProvision] === "true") {
+      Object.assign(configDev, { auth: {} });
+      if (envDefault[ResourcePlugins.Aad][EnvConfigName.OAuthScopeId]) {
+        Object.assign(configDev.auth, {
```
Comment (blackchoey) [inline packages/fx-core/src/core/middleware/projectMigrator.ts:492]: can we use `configDev.auth.accessAsUserScopeId = xxx` to align the coding style with other part (e.g. https://github.com/OfficeDev/TeamsFx/blob/dev/packages/fx-core/src/core/middleware/projectMigrator.ts#L858)?

File: packages/fx-core/tests/core/hooks.test.ts (hunk 1 above)
Comment (wenytang-ms) [reply packages/fx-core/tests/core/hooks.test.ts]: fixed, add ignoreEnvInfo to inputs as the common load logic

File: packages/fx-core/src/core/middleware/projectMigrator.ts [hunk 3]
```diff
@@ -85,6 +85,7 @@ const parameterFileNameTemplate = "azure.parameters.@envName.json";
 const learnMoreLink = "https://aka.ms/teamsfx-migration-guide";
 const manualUpgradeLink = `${learnMoreLink}#upgrade-your-project-manually`;
 const upgradeReportName = `upgrade-change-logs.md`;
+const AadSecret = "{{ $env.AAD_APP_CLIENT_SECRET }}";
```
[the remark this answers is not in the document]
Comment (dooriya) [reply packages/fx-core/src/core/middleware/projectMigrator.ts:90]: The naming looks good to me~

**Review by blackchoey — APPROVED** [review]

**Pull request merged**